micrometer-test/src/test/java/io/micrometer/core/instrument/JerseyServerTimingInstrumentationVerificationTests.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument;
+
+import io.micrometer.core.instrument.binder.jersey.server.DefaultJerseyTagsProvider;
+import io.micrometer.core.instrument.binder.jersey.server.MetricsApplicationEventListener;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+
+class JerseyServerTimingInstrumentationVerificationTests extends HttpServerTimingInstrumentationVerificationTests {
+
+    JerseyTest jerseyTest;
+
+    @Override
+    protected URI startInstrumentedServer() throws Exception {
+        jerseyTest = new JerseyTest() {
+            @Override
+            protected Application configure() {
+                MetricsApplicationEventListener listener = new MetricsApplicationEventListener(getRegistry(),
+                        new DefaultJerseyTagsProvider(), timerName(), true);
+
+                ResourceConfig config = new ResourceConfig();
+                config.register(listener);
+                config.register(TestResource.class);
+
+                return config;
+            }
+        };
+
+        jerseyTest.setUp();
+
+        return jerseyTest.target().getUri();
+    }
+
+    @Override
+    protected void stopInstrumentedServer() throws Exception {
+        jerseyTest.tearDown();
+    }
+
+    @Path("/")
+    @Produces(MediaType.TEXT_PLAIN)
+    public static class TestResource {
+
+        @GET
+        public String root() {
+            return "hello";
+        }
+
+        @GET
+        @Path("hello/{name}")
```

Comment:
I wonder if these strings can't be  static finals in the test abstract class. That way you will be sure that you're using the proper strings. WDYT?

Reply:
I agree it would be good to have some way to avoid typos and improve discoverability beyond the current setup of it being in the JavaDoc. I'm wondering if an enum might make it easy for implementors to find all the routes that should be implemented. I'm going to draft something and see if I can get some feedback on it.

Reply:
Ah, I tried making the enum and immediately realized that won't work for using here because the annotation needs a constant value, which is doesn't recognize enum fields/methods as constant, even though they effectively are.

Reply:
See newly added `InstrumentedRoutes` class with string constants. WDYT?